tests/ballerina-test-integration/src/test/java/org/ballerinalang/test/security/HTTPResponseXMLSecurityTestCase.java

```diff
@@ -0,0 +1,103 @@
+                + File.separator + "xmlSecurity" + File.separator +
+                "xml-parsing-service.bal").getAbsolutePath();
+        ballerinaServer = ServerInstance.initBallerinaServer();
+        ballerinaServer.startBallerinaServer(balFilePath);
+    }
+
+    @Test(description = "Test the service for XML External Entity Injection attack")
+    public void testXMLExternalEntityInjection() throws Exception {
+        String xmlString = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" +
+                "<!DOCTYPE foo [" +
+                "<!ELEMENT foo ANY >" +
+                "<!ENTITY xxe SYSTEM \"https://www.w3schools.com/xml/note.xml\" >]>" +
+                "<foo>&xxe;</foo>";
+        HttpResponse response = sendXmlPayload(xmlString);
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+        String respMsg = "";
+        Assert.assertEquals(response.getData(), respMsg, "Message content mismatched");
+    }
+
+    @Test(description = "Test the service for XML Entity Expansion attack", timeOut = 10000)
+    public void testXMLEntityExpansion() throws Exception {
+        String xmlString = "<?xml version=\"1.0\"?>\n" +
+                "<!DOCTYPE lolz [\n" +
+                "<!ENTITY lol \"lol\">\n" +
+                "<!ENTITY lol2 \"&lol;&lol;&lol;&lol;&lol;&lol;&lol;&lol;&lol;&lol;\">\n" +
+                "<!ENTITY lol3 \"&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;\">\n" +
+                "<!ENTITY lol4 \"&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;\">\n" +
+                "<!ENTITY lol5 \"&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;\">\n" +
+                "<!ENTITY lol6 \"&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;\">\n" +
+                "<!ENTITY lol7 \"&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;\">\n" +
+                "<!ENTITY lol8 \"&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;\">\n" +
+                "<!ENTITY lol9 \"&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;\">\n" +
+                "]>\n" +
+                "<lolz>&lol9;</lolz>";
+        HttpResponse response = sendXmlPayload(xmlString);
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), 500, "Response code mismatched");
+        //TODO: Introduce validation of error message once GitHub issue #4533 is fixed.
+    }
+
+    private HttpResponse sendXmlPayload(String requestMsg) throws Exception {
+        Map<String, String> headers = new HashMap<>();
+        headers.put(HttpHeaderNames.CONTENT_TYPE.toString(), TestConstant.CONTENT_TYPE_XML);
+        String serviceUrl = "http://localhost:9090/xmlparser";
+        HttpResponse response = HttpClientRequest.doPost(serviceUrl, requestMsg, headers);
+        if (response == null) {
+            //Retrying to avoid intermittent test failure
```

Comment:
Do we know the cause for this intermittent issue?

Reply:
This is from other similar integration tests. I do not want to introduce any  intermittent failures, hence I have followed same patterns other similar tests have used. Please check [1]. Reasoning or further analysis of this should be done though std-lib team if that is necessary at this point. 

[1] https://github.com/ballerina-platform/ballerina-lang/search?q=%22Retrying+to+avoid+intermittent+test+failure%22&unscoped_q=%22Retrying+to+avoid+intermittent+test+failure%22